product-service/src/main/java/com/extremecoder/productservice/utils/UrlConstraint.java

```diff
@@ -0,0 +1,17 @@
+package com.extremecoder.productservice.utils;
+
+public final class UrlConstraint {
+
+    private UrlConstraint() {
+    }
+
+    private static final String API = "/api";
+    private static final String VERSION = "/v1";
+
+    public static class ProductManagement {
+        public static final String ROOT = API + VERSION + "/products";
+        public static final String DELETE = "/{productId}";
+        public static final String GET = "/{productId}";
+        public static final String PUT = "/{productId}";
```

Comment:
Here Both PUT and GET is assigning same string, which we could improve. 
- We could use only one variable as an example `PARAM_PROD_ID` and use it in controller layer.

Reply:
Can't get your point. Could you please elaborate this ?

Reply:
```
        public static final String DELETE = "/{productId}";
        public static final String GET = "/{productId}";
        public static final String PUT = "/{productId}";
```
All these variables are assigning same string `/{productId}` which is redundant. We writing more code here which is against clean code convention. 
- Instead you could keep a variable like  `PARAM_PROD_ID` and use it in the controller layer. The variable name can be anything which matches best with the context.